src/core/particle_data.cpp

```diff
@@ -316,9 +322,10 @@ Utils::Cache<int, Particle> particle_fetch_cache(max_cache_size);
 void invalidate_fetch_cache() { particle_fetch_cache.invalidate(); }
 
 const Particle *get_particle_data(int part) {
-  auto const pnode = get_particle_node(part);
-  /* Check if particle exists at all. */
-  if (-1 == pnode) {
+  int pnode;
```

Comment:
If you want to be consequent, you should do
```c++
return particle_exists(part) ? get_particle_node(part) : nullptr;
```
which is somewhat more idiomatic, but results in two lookups. Alternatively you can get the iterator and compare it to end.